tutor/src/components/add-edit-question/ux.js

```diff
@@ -530,10 +530,12 @@ export default class AddEditQuestionUX {
         const exercise = this.from_exercise_id ? this.exercises.get(this.from_exercise_id) : null;
 
         await this.exercises.createExercise(this.toFormData).then(() => {
-            this.clearAutosave();
-            this.disableAutosave();
-            this.changed = false;
-            this.enableAutosave();
+            runInAction(() => {
+                this.clearAutosave();
+                this.disableAutosave();
+                this.changed = false;
+                this.enableAutosave();
+            })
         });
```

Comment:
this works fine, but you can also wrap functions in an action:
```js
await this.exercises.createExercise(this.toFormData).then(action(() => {
   this.clearAutosave();
                 this.disableAutosave();
                 this.changed = false;
                 this.enableAutosave();
})
```

Reply:
Oh wow I like that pattern a lot better 👍